src/cmd/services/m3dbnode/main/main_test.go

```diff
@@ -616,3 +634,42 @@ db:
                   endpoint: {{.InitialClusterEndpoint}}
 `
 )
+
+// waitUntilAllShardsAreAvailable continually polls the session checking to see if the topology.Map
+// that the session is currently storing contains a non-zero number of host shard sets, and if so,
+// makes sure that all their shard states are Available.
+func waitUntilAllShardsAreAvailable(t *testing.T, session client.AdminSession) {
+	for {
+		topoMap, err := session.TopologyMap()
+		require.NoError(t, err)
+
+		var (
+			hostShardSets         = topoMap.HostShardSets()
+			allShardsAreAvailable = true
+		)
+
+		if len(hostShardSets) == 0 {
+			// We haven't received an actual topology yet.
+			continue
+		}
+
+		for _, hostShardSet := range hostShardSets {
+			for _, hostShard := range hostShardSet.ShardSet().All() {
+				if hostShard.State() != shard.Available {
+					allShardsAreAvailable = false
+					break
+				}
+			}
+
+			if !allShardsAreAvailable {
+				break
+			}
+		}
+
+		if allShardsAreAvailable {
+			break
```

Comment:
Is `break`ing multiple levels like this clearer or via [label scopes](https://golang.org/ref/spec#Break_statements)?